src/anomalib/engine/engine.py

```diff
@@ -127,7 +127,7 @@ def __init__(
         | ListConfig
         | list[dict[str, str | float]]
         | str = "F1AdaptiveThreshold",
-        task: TaskType = TaskType.SEGMENTATION,
+        task: TaskType | str = TaskType.SEGMENTATION,
```

Comment:
just wondering if we should call this `task_type` instead of `task`, as `task` could also refer to an entity that actually performs some learning or inference related behaviour (as we see in some other libraries). `task_type` is less ambiguous.